Sources/performance-worker/RPCStats.swift

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import NIOConcurrencyHelpers
+
+/// Stores the real time latency histogram and error code count dictionary,
+/// for the RPCs made by a particular GRPCClient. It gets updated after
+/// each finished RPC.
+struct RPCStats {
+  var latencyHistogram: LatencyHistogram
+  var requestResultCount: [Int32: Int64]
```

Comment:
Any reason we can't use `Status.Code` as the key?

Reply:
not necessarily, the output message representing this counts uses an Int32 for the statusCode, but this doesn't mean the transformation can't be done exactly when creating the response, so I will transform the keys of this dictionary to be of Status.Code type

Reply:
Given the comments above about error vs. status, we should actually use `RPCError.Code` instead. We should use the "right" types from an API perspective and then convert to whatever the proto wants when we populate it later.